src/app/search-bar/search-bar.component.ts

```diff
@@ -113,6 +115,9 @@ export class SearchBarComponent implements OnInit, AfterViewInit {
 
   ngOnInit() {
     this.displayStatus = 'hidebox';
+    this.boxWidth = document.getElementById('nav-group').style.width.replace('px', '');
```

Comment:
in that case, I would prefer writing two different classes A and B in search bar component and use class A when it is in the index page and class B when it is in the results page.In this way, we could maintain styles from just CSS but change them dynamically using javascript.